src/webui/abstractwebapplication.cpp

```diff
@@ -310,10 +310,15 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
+    return ((!env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("10.0.0.0/8"))
+            && !env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("172.16.0.0/12"))
+            && !env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("192.168.0.0/16"))
+            && !env_.clientAddress.isInSubnet(QHostAddress::parseSubnet("fc00::/7")))
+            || Preferences::instance()->isWebUiLANAuthEnabled())
+            && ((env_.clientAddress != QHostAddress::LocalHost
```

Comment:
There are many conditions here so it would be better to have two helper functions: isLoopbackAddress and isLANAddress.